From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Danny Lin <danny@kdrag0n.dev>
Date: Sat, 9 Oct 2021 14:37:44 -0700
Subject: [PATCH 02/14] overlay: Offload WM shell to another thread

Android 12 added this config to offload some window management to a
dedicated thread. Enable it to improve performance when the main System
UI thread is busy.

Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 .../WindowManager/Shell/res/values/config.xml | 22 +++++++++++++++++++
 1 file changed, 22 insertions(+)
 create mode 100644 overlay/common/frameworks/libs/WindowManager/Shell/res/values/config.xml

diff --git a/overlay/common/frameworks/libs/WindowManager/Shell/res/values/config.xml b/overlay/common/frameworks/libs/WindowManager/Shell/res/values/config.xml
new file mode 100644
index 0000000..3f6009a
--- /dev/null
+++ b/overlay/common/frameworks/libs/WindowManager/Shell/res/values/config.xml
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+     Copyright (C) 2023 crDroid Android Project
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+<resources>
+
+    <!-- Determines whether the shell features all run on another thread. This is to be overrided
+         by the resources of the app using the Shell library. -->
+    <bool name="config_enableShellMainThread">true</bool>
+</resources>
